big_scape/distances/query.py

```diff
@@ -159,47 +165,56 @@ def calculate_distances(run: dict, bin: bs_comparison.RecordPairGenerator):
         # fetches the current number of singleton ref <-> connected ref pairs from the database
         num_pairs = bin.num_pairs()
 
-        # if there are no more singleton ref <-> connected ref pairs, then break and exit
-        if num_pairs == 0:
-            break
-
-        logging.info("Calculating distances for %d pairs", num_pairs)
+        if num_pairs > 0:
```

Comment:
possible to do a 
```
if num_pairs == 0:
   something something
   continue
```

to reduce nesting? OK if not or time-consuming.

Reply:
I tried at the start, but you end up duplicating the break/cycle checks that occur later on.. So I'd say this is fairly clean for now.